### ISSUE TITLE
Naive implementation in plugin system

### ISSUE THREAD

**Issue by sillydan1** [sillydan1/AALTITOAD#40]
The function `bool is_dynamic_library(const std::string& filename)` implementation is quite naïve in the sense that it just checks if the filename _contains_ dylib/so/dll. 

This means that a file named `foo.so.txt` is considered a dynamic library and is the program attempts to load it as a plug-in. 

A real solution would be to use magic

**Comment by kiner-shah** [comment]
Terminfo? Wouldn't that give information about terminal?

**Comment by sillydan1** [comment]
> Terminfo? Wouldn't that give information about terminal?

Correct, I think I meant the 'file' command, or maybe https://en.cppreference.com/w/cpp/filesystem/file_type or similar

**Comment by kiner-shah** [comment]
Which file is this function in?

**Comment by sillydan1** [comment]
> Which file is this function in?

Right, sorry. I was referring to the [Linux file command](https://www.man7.org/linux/man-pages/man1/file.1.html) :smile: I'm not sure if it is available on Windows though - and I'm also not sure which library it is part of :thinking:

**Comment by kiner-shah** [comment]
I meant, which file is `bool is_dynamic_library(const std::string& filename)` in?

**Comment by sillydan1** [comment]
It's in the [`plugin_system.cpp`](https://github.com/search?q=repo%3Asillydan1%2Faaltitoad%20is_dynamic_library&type=code) :smile:

**Comment by kiner-shah** [comment]
Maybe the check for `is_dynamic_library` isn't required. If `dlopen` fails, then it fails; instead of throwing a logic error, better to log a message "couldn't load file as shared/dynamic lib" and continue.

**Comment by sillydan1** [comment]
So, just remove https://github.com/sillydan1/aaltitoad/blob/7ff31bb39d8e5a760e455142a945295e05e04d44/src/plugin_system/plugin_system.cpp#L52 ? Yea, that probably works just fine 🤔 

I might also downgrade the warning log to an info level, just so you dont get weird warnings about being unable to load non-plugin files.

**Comment by kiner-shah** [comment]
You may keep it at debug level actually, if a plugin is loaded, then a info level log would make sense.
